lms/djangoapps/discussion/tasks.py

```diff
@@ -0,0 +1,92 @@
+"""
+Defines asynchronous celery task for sending email notification (through edx-ace)
+pertaining to new discussion forum comments.
+"""
+import logging
+from urlparse import urljoin
+
+from celery import task
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib.sites.models import Site
+
+from celery_utils.logged_task import LoggedTask
+from edx_ace import ace
+from edx_ace.message import MessageType
+from edx_ace.recipient import Recipient
+from opaque_keys.edx.keys import CourseKey
+from lms.djangoapps.django_comment_client.utils import permalink
+import lms.lib.comment_client as cc
+from lms.lib.comment_client.utils import merge_dict
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.schedules.template_context import get_base_template_context
+
+
+log = logging.getLogger(__name__)
+
+
+DEFAULT_LANGUAGE = 'en'
+ROUTING_KEY = getattr(settings, 'ACE_ROUTING_KEY', None)
+
+
+class ResponseNotification(MessageType):
+    def __init__(self, *args, **kwargs):
+        super(ResponseNotification, self).__init__(*args, **kwargs)
+        self.name = 'response_notification'
+
+
+@task(base=LoggedTask, routing_key=ROUTING_KEY)
+def send_ace_message(context):
+    context['course_id'] = CourseKey.from_string(context['course_id'])
+
+    if _should_send_message(context):
+        thread_author = User.objects.get(id=context['thread_author_id'])
+        message_context = _build_message_context(context)
+        message = ResponseNotification().personalize(
+            Recipient(thread_author.username, thread_author.email),
+            _get_course_language(context['course_id']),
+            message_context
+        )
+        log.info('Sending forum comment email notification with context %s', message_context)
+        ace.send(message)
+
+
+def _should_send_message(context):
```

Comment:
Could you just pass in course_id rather than the entire context?

Reply:
There's three pieces of information we need right now: course_id, thread_author_id, and thread_id.  We'll also need the comment_id in the very near future.

Reply:
Oh, right. I don't know why I missed that.